tools/SetupFlow/DevHome.SetupFlow/ViewModels/AddRepoViewModel.cs

```diff
@@ -298,12 +328,127 @@ private void CancelButtonPressed()
+        }
+
+        newMenu.Items.Add(new MenuFlyoutSeparator());
+        var thatMenuItem = new MenuFlyoutItem();
+        thatMenuItem.Text = _stringResource.GetLocalized("RepoToolAddAnotherAccount");
+        thatMenuItem.Click += AddAccountClicked;
+        newMenu.Items.Add(thatMenuItem);
+
+        return newMenu;
+    }
+
+    /// <summary>
+    /// The bottom of the MenuFlyout has a button to log into another account.  Handle logging the user in.
+    /// </summary>
+    /// <param name="sender">The object that sent this event.</param>
+    /// <param name="e">The arguments</param>
+    /// <remarks>
+    /// This calls MenuItemClick to poulate the list of repos if a new account is detected.
+    /// </remarks>
+    private async void AddAccountClicked(object sender, RoutedEventArgs e)
+    {
+        if (sender is MenuFlyoutItem selectedItem &&
+            selectedItem.Text.Equals(_stringResource.GetLocalized("RepoToolAddAnotherAccount"), StringComparison.OrdinalIgnoreCase))
+        {
+            // If the user selects repos from account 1, then logs into account 2 and does not save between those two actions
+            // _previouslySelectedRepos will be empty.  The result is the repos in account 1 will not be selected if the user navigates
+            // to account 1 after logging into account 2.
+            // Save the repos here in that case.
+            if (!_previouslySelectedRepos.Any())
+            {
+                _previouslySelectedRepos.AddRange(EverythingToClone);
+            }
+
+            ShowRepoPage = Visibility.Collapsed;
+
+            // Store the logged in accounts to help figure out what account the user logged into.
+            var loggedInAccounts = await Task.Run(() => _providers.GetAllLoggedInAccounts(_selectedRepoProvider));
+            await LogUserIn(_selectedRepoProvider, _addRepoDialog.GetLoginUiContent(), true);
+            var loggedInAccountsWithNewAccount = await Task.Run(() => _providers.GetAllLoggedInAccounts(_selectedRepoProvider));
+
+            ShowRepoPage = Visibility.Visible;
+            Accounts = new ObservableCollection<string>(loggedInAccountsWithNewAccount.Select(x => x.LoginId));
+            AccountsToShow = ConstructFlyout();
+
+            // The dialog makes a user log in if they have no accounts.
+            // keep this here just in case.
+            if (Accounts.Any())
+            {
+                var newAccount = loggedInAccountsWithNewAccount.Except(loggedInAccounts);
+
+                // Logging in should allow only one account to log in at a time.
+                if (newAccount.Count() > 1)
+                {
+                    Log.Logger?.ReportError(Log.Component.RepoConfig, $"{newAccount.Count()} accounts logged in at once.  Choosing the first alphabetically");
+                }
+
+                if (newAccount.Any())
+                {
+                    SelectedAccount = newAccount.OrderByDescending(x => x.LoginId).FirstOrDefault().LoginId;

```

Comment:
Same comment about updating LoginId to URL

Reply:
Made a bug to track this change.